File: src/main/kotlin/game/entity/Rank.kt
package eelst.ilike.game.entity

enum class Rank(val numericalValue: Int) : CardAttribute {
    ONE(1),
    TWO(2),
    THREE(3),
    FOUR(4),
    FIVE(5);

    companion object {
        fun getByNumericalValue(numericalValue: Int): Rank {
            require(numericalValue > 0 && numericalValue <= 5) {
                "No Rank with numerical value equal to $numericalValue"
            }
            return entries.first { it.numericalValue == numericalValue }
        }
    }
}

File: src/game/rank.rs
use std::fmt;
#[derive(PartialEq, Debug)]
pub enum Rank {
    Start,
    One,
    Two,
    Three,
    Four,
    Five,
}

impl fmt::Display for Rank {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        match self {
            Rank::Start => write!(f, "START"),
            Rank::One => write!(f, "1"),
            Rank::Two => write!(f, "2"),
            Rank::Three => write!(f, "3"),
            Rank::Four => write!(f, "4"),
            Rank::Five => write!(f, "5"),
        }
    }
}
